.github/workflows/python-build-test.yaml

```diff
@@ -211,24 +211,34 @@ jobs:
         run: pytest pylace/tests
 
       - name: Upload wheels
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: wheels
+          name: wheels-macos-${{ matrix.runs-on }}
           path: dist
 
+  merge:
+    runs-on: ubuntu-latest
+    needs: [ macos, windows, linux ]
+    steps:
+      - name: Merge Artifacts
+        uses: actions/upload-artifacts/merge@v4
+        with:
+          name: wheels
+          pattern: wheels-*
+
   test-mdbook-python:
     name: Test MDBook Python Snippets
     runs-on: ubuntu-latest
     needs: [linux]
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
         with:
           name: wheels
           path: dist
```

Comment:
If I'm reading the [merge docs](https://github.com/actions/upload-artifact/blob/main/merge/README.md) correctly, it seems like it's preferred to just use the `pattern` and `merge-multiple` flags for `download-artifact` in the new system? But I don't know for certain.

Reply:
I followed the current issue for this from the devs. We also want to merge the artifacts without the publish step.

Reply:
If you don't like the current pattern, I can change it.

Reply:
No I'm not opposed. I already approved it, it's good by me.